linked_list/linked_list.py

```diff
@@ -23,13 +23,57 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
+    exists = self.head
+    while(exists):
+      exists = exists.next_node
+    exists = value
+    self.tail = value
+    return True
 
   def remove_head(self):
-    pass
+    # if head exists go on
+      # 
+      if self.head:
+        if self.head.get_value() == None:
+          temp_head = self.head
+
+          self.head = None
+          self.tail = None
+
+          return temp_head.get_value()
+        else:
+
+    # else return None
 
   def contains(self, value):
-    pass
+    curr_node = self.head
+    while(curr_node):
+      if curr_node.value == value:
+        return True
+      else:
+        curr_node = curr_node.next_node
```

Comment:
What happens if you get to tail of the list and still haven't encountered a node with the value?